docs/writing/posts/lgtmk.md

```diff
@@ -0,0 +1,229 @@
+---
+
+# Stop using LGTM@Few as a metric (Better RAG)
+
+!!! warning "This is a draft"
+
+    This is a draft. I would love to receive any PRs that correct any typos. Thanks.
+
+When giving [advice](./rag.md) to developers on [improving](./rag-inverted.md) their retrieval augmented generation, I usually say two things:
+
+1. Look at the Data
+2. Don't just look at the Data
+
+Wise men speak in paradoxes because we are afraid of half-truths. This blog post will try to capture when to look at data and when to stop looking at data in the context of retrieval augmented generation.
+
+I'll cover the different relevancy and ranking metrics, some stories to help you understand them, their trade-offs, and some general advice on how to think.
+
+<!-- more -->
+
+## When to look at data?
+
+Look at data when the problem is very new. Do not rely on any kinds of metrics just yet. Look at the queries people are asking. Look at the documents that people are submitting. Look at the text chunks and see whether or not a single text chunk could possibly answer a question your user might have, or if you need multiple text chunks to piece together a complete answer. Look at the results from initial prototypes to understand if the retrieval task is technically feasible.
+
+There's a lot of intuition you can gain from just looking at the data.
+
+## When to stop?
+
+At some point, you're going to actually want to build a system. You're going to want to iterate and improve on it. You will likely get nowhere if all you're doing is 'looking at things'. You will spend too much time guessing as to what will improve something rather than trying to measure and improve something.
+
+> "What gets measured gets managed."
+
+Instead, define metrics, run tests, investigate when and where the metrics are poor, and then start looking at the data again.
+
+```mermaid
+graph LR
+    A[Look at Data];
+    A --> B[Work on System]
+    B --> C[Define Metrics]
+    C --> D[Look at Metrics]
+    D --> E[Look at Data with Poor Metrics]
+    E --> B
+```
+
+Well, let's take a closer look at what kind of metrics we can use and how they might improve our system. And I'll give an intuitive understanding of why and how some of these metrics break down. But first I also want to talk about the importance of speed.
+
+## Importance of Velocity
+
+How quickly you can get metrics and run tests determines the nature of how you iterate on your software. If you're looking at a metric that takes a long time to compute, you're going to be waiting a long time to iterate on your system. So do whatever it takes to make the test that you run and the metrics you build as fast as possible!
+
+**Example via RAG**
+
+- **Slow Metric**: Collocating human preferences and consulting domain experts.
+- **Still Slow Metric**: AI-generated metrics. When using something like GPT4, things can become very slow.
+- **Fast Metrics**: Accuracy, Precision, Recall, MRR, NDCG, are computationally cheap given the labels.
```

Comment:
for RAG, it's usually hard to collect rank-aware data. users might not click on the references/citation links at all.  so mrr and ndcg will be hard to infer.